dialer.go

```diff
@@ -446,12 +463,19 @@ func (d *Dialer) Dial(ctx context.Context, icn string, opts ...DialOption) (conn
 	}
 	return iConn, nil
 }
+func (d *Dialer) isTLSError(err error) bool {
+	if nErr, ok := err.(net.Error); ok {
+		return !nErr.Timeout() && // it's a permanent net error
```

Comment:
Have you manually tested this logic as part of doing a cert rotation to confirm it's working as intended?

Reply:
Might be worth adding an e2e test for this, but otherwise this seems fine.

Reply:
Ah got it. I missed that detail. I agree then -- no e2e is fine.